### PR TITLE
CSOT: Adjust timeouts and increase test coverage

### PR DESCRIPTION
**Pull request by vbabanin**
Description

This PR introduces the following changes:

- Timeout Adjustments: Some timeouts have been reduced to ensure tests pass in serverless configurations.
- Additional Prose Tests: Added new prose tests to cover more scenarios.
- Re-enabled ignored tests: Uncommented and re-enabled previously ignored tests.

JAVA-5104

### CODE REVIEW

File: driver-core/src/test/resources/unified-test-format/client-side-operation-timeout/sessions-override-operation-timeoutMS.json [hunk 1]
```diff
@@ -97,9 +100,6 @@
         {
           "name": "commitTransaction",
           "object": "session",
-          "arguments": {
-            "timeoutMS": 500
```
Comment (rozza) [inline driver-core/src/test/resources/unified-test-format/client-side-operation-timeout/sessions-override-operation-timeoutMS.json:101]: Whats the reasoning for this change?

File: driver-sync/src/test/functional/com/mongodb/client/AbstractClientSideOperationsTimeoutProseTest.java [hunk 2]
```diff
@@ -742,10 +732,53 @@ public void shouldIgnoreWtimeoutMsOfWriteConcernToInitialAndSubsequentCommitTran
...
-                assertFalse(writeConcern.containsKey("wTimeoutMS"));
+                assertFalse(writeConcern.containsKey("wtimeout"));
             }});
     }
 
+    /**
+     * Not a prose spec test. However, it is additional test case for better coverage.
+     */
+    @Test
+    @DisplayName("Should refresh timeout for commit transaction")
```
Comment (rozza) [inline driver-sync/src/test/functional/com/mongodb/client/AbstractClientSideOperationsTimeoutProseTest.java:743]: How does this test check that the timeout is refreshed?  Seems to just confirm that the second commit fail with MongoOperationTimeoutException?

File: driver-core/src/test/resources/unified-test-format/client-side-operation-timeout/non-tailable-cursors.json [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
   "description": "timeoutMS behaves correctly for non-tailable cursors",
+  "comment": "Manually reduced blockTimeMS for tests to pass in serverless",
```
Comment (rozza) [inline driver-core/src/test/resources/unified-test-format/client-side-operation-timeout/non-tailable-cursors.json:3]: Wowzers

File: driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/ClientSideOperationTimeoutProseTest.java [hunk 4]
```diff
@@ -107,7 +105,6 @@ protected boolean isAsync() {
         return true;
     }
 
-    @Tag("setsFailPoint")
```
Comment (rozza) [inline driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/ClientSideOperationTimeoutProseTest.java:110]: 👍

**Review by rozza — COMMENTED** [review]
Some questions

File: driver-core/src/test/resources/unified-test-format/client-side-operation-timeout/sessions-override-operation-timeoutMS.json (hunk 1 above)
Comment (vbabanin) [reply driver-core/src/test/resources/unified-test-format/client-side-operation-timeout/sessions-override-operation-timeoutMS.json]: I have reverted this change as we don't have overrides. Thank you for pointing it out!

File: driver-sync/src/test/functional/com/mongodb/client/AbstractClientSideOperationsTimeoutProseTest.java (hunk 2 above)
Comment (vbabanin) [reply driver-sync/src/test/functional/com/mongodb/client/AbstractClientSideOperationsTimeoutProseTest.java]: I have corrected the display name to be `Should throw timeout exception for subsequent commit transaction`. Good catch!

File: driver-sync/src/test/functional/com/mongodb/client/AbstractClientSideOperationsTimeoutProseTest.java [hunk 5]
```diff
@@ -742,10 +732,53 @@ public void shouldIgnoreWtimeoutMsOfWriteConcernToInitialAndSubsequentCommitTran
             if (command.containsKey("writeConcern")) {
                 BsonDocument writeConcern = command.getDocument("writeConcern");
                 assertFalse(writeConcern.isEmpty());
-                assertFalse(writeConcern.containsKey("wTimeoutMS"));
+                assertFalse(writeConcern.containsKey("wtimeout"));
```
Comment (rozza) [inline driver-sync/src/test/functional/com/mongodb/client/AbstractClientSideOperationsTimeoutProseTest.java:735]: 👍

**Review by rozza — APPROVED** [review]
LGTM!

**Pull request merged**